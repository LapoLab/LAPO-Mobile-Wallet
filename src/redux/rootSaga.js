import { all } from 'redux-saga/effects';
import commonsSagas from '../commons/sagas';
import recoveryWordsSagas from '../modules/RecoveryWords/sagas';
import sendSagas from '../modules/Send/sagas';
import paperWordsConfirm from '../modules/PaperWordsConfirm/sagas';
import dashboard from '../modules/Dashboard/sagas';
import settings from '../modules/Settings/sagas';
import paperWords from '../modules/PaperWords/sagas';
import transactionsDetails from '../modules/TransactionDetails/sagas';
import chooseContact from '../modules/ChooseContact/sagas';
import splashScreen from '../modules/SplashScreen/sagas';
import switchContacts from '../modules/SwitchContacts/sagas';
import receive from '../modules/Receive/sagas';

// eslint-disable-next-line
const getListeners = (...args) => args.reduce((acc, nextArg) => [...acc, ...Object.values(nextArg).map(func => func())], []);

export default function* rootSaga() {
  yield all(
    getListeners(
      commonsSagas,
      recoveryWordsSagas,
      sendSagas,
      paperWordsConfirm,
      dashboard,
      settings,
      paperWords,
      transactionsDetails,
      chooseContact,
      splashScreen,
      switchContacts,
      receive,
    ),
  );
}
