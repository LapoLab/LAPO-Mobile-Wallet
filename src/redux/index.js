import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';
import storage from 'redux-persist/lib/storage';
import { persistStore, persistReducer } from 'redux-persist';
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2';
import rootReducer from './rootReducer';
import rootSagas from './rootSaga';

const persistConfig = {
  key: 'root',
  storage,
  stateReconciler: autoMergeLevel2,
  blacklist: [
    'modals',
    'dashboard',
    'paperWords',
    'chooseContact',
    'send',
    'transactionsDetails',
    'alert',
    'pinsRecovery',
  ],
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

const sagaMiddleware = createSagaMiddleware();
// eslint-disable-next-line no-undef
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export const store = createStore(
  persistedReducer,
  composeEnhancers(applyMiddleware(sagaMiddleware)),
);
sagaMiddleware.run(rootSagas);

export const persistor = persistStore(store);
