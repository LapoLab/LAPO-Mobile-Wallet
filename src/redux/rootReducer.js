import { combineReducers } from 'redux';

import wallet from '../commons/reducers/wallet';
import pins from '../commons/reducers/pins';
import pinsRecovery from '../commons/reducers/pinsRecovery';
import modals from '../commons/reducers/modals';
import settings from '../commons/reducers/settings';
import notify from '../modules/Notify/reducers';
import paperConfirm from '../modules/PaperWordsConfirm/reducer';
import dashboard from '../modules/Dashboard/reducers';
import paperWords from '../modules/PaperWords/reducers';
import chooseContact from '../modules/ChooseContact/reducers';
import send from '../modules/Send/reducers';
import transactionsDetails from '../modules/TransactionDetails/reducer';
import alert from '../modules/Alert/reducers';

export default combineReducers({
  wallet,
  pins,
  pinsRecovery,
  notify,
  modals,
  settings,
  paperConfirm,
  dashboard,
  paperWords,
  chooseContact,
  send,
  transactionsDetails,
  alert,
});
