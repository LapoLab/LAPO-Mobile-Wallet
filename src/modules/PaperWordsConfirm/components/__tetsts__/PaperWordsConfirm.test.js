// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { store } from '../../../../redux';
import { PaperWordsConfirm } from '../index';

describe('PaperWordsConfirm page', () => {
  test('renders snapshots', () => {
    const navigation = {
      addListener: jest.fn(),
      dispatch: jest.fn(),
      goBack: jest.fn(),
    };

    const tree = renderer
      .create(
        <Provider store={store}>
          <PaperWordsConfirm
            navigation={navigation}
            index1={2}
            index2={8}
            confirmWord={jest.fn()}
            genIndexForWords={jest.fn()}
          />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
