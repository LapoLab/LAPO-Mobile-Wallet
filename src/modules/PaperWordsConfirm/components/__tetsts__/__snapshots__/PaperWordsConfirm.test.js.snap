// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaperWordsConfirm page renders snapshots 1`] = `
<BVLinearGradient
  colors={
    Array [
      -12483651,
      -16172276,
    ]
  }
  endPoint={
    Object {
      "x": 1.2,
      "y": 1.2,
    }
  }
  locations={
    Array [
      0,
      1,
    ]
  }
  startPoint={
    Object {
      "x": -0.1,
      "y": -0.1,
    }
  }
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
          "justifyContent": "space-between",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 30,
            "justifyContent": "flex-start",
            "marginTop": 20,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/back.png",
              }
            }
            style={
              Object {
                "flex": 1,
                "marginLeft": 12,
                "resizeMode": "contain",
                "width": 12,
              }
            }
          />
        </View>
        <Text
          style={
            Object {
              "alignItems": "center",
              "color": "white",
              "flex": 1,
              "fontFamily": "Lato-Regular",
              "fontSize": 22,
              "marginRight": 24,
              "textAlign": "center",
            }
          }
        >
          Confirm words
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 50,
            "justifyContent": "center",
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/indicator4.png",
            }
          }
          style={
            Object {
              "height": 20,
              "resizeMode": "contain",
            }
          }
        />
      </View>
      <RCTScrollView>
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 120,
                "justifyContent": "center",
                "paddingHorizontal": 25,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontFamily": "Lato-Light",
                  "fontSize": 20,
                  "textAlign": "center",
                }
              }
            >
              To make sure everything was written down correctly, please enter the following words from your paper key.
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "column",
                "padding": 25,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "column",
                  "marginBottom": 20,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontFamily": "Lato-Regular",
                    "fontSize": 20,
                    "marginTop": 10,
                  }
                }
              >
                Word #2
              </Text>
              <TextInput
                allowFontScaling={true}
                onChangeText={[Function]}
                style={
                  Object {
                    "borderBottomWidth": 1,
                    "borderColor": "white",
                    "color": "white",
                    "fontFamily": "Lato-Regular",
                    "fontSize": 20,
                    "height": 40,
                    "marginTop": 10,
                    "paddingBottom": 5,
                    "width": "100%",
                  }
                }
                underlineColorAndroid="transparent"
              />
            </View>
            <View
              style={
                Object {
                  "flexDirection": "column",
                  "marginBottom": 20,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontFamily": "Lato-Regular",
                    "fontSize": 20,
                    "marginTop": 10,
                  }
                }
              >
                Word #8
              </Text>
              <TextInput
                allowFontScaling={true}
                onChangeText={[Function]}
                style={
                  Object {
                    "borderBottomWidth": 1,
                    "borderColor": "white",
                    "color": "white",
                    "fontFamily": "Lato-Regular",
                    "fontSize": 20,
                    "height": 40,
                    "marginTop": 10,
                    "paddingBottom": 5,
                    "width": "100%",
                  }
                }
                underlineColorAndroid="transparent"
              />
            </View>
          </View>
        </View>
      </RCTScrollView>
      <View
        style={
          Object {
            "height": 100,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#4183BD",
                "borderColor": "rgb(42, 101, 117)",
                "borderRadius": 74,
                "borderWidth": 8,
                "height": 64,
                "justifyContent": "center",
                "marginHorizontal": 25,
                "opacity": 1,
                "width": "80%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontFamily": "Lato-Regular",
                  "fontSize": 18,
                }
              }
            >
              VERIFY IT!
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</BVLinearGradient>
`;
